coffee_maker/code_formatter/crewai/main.py

```diff
@@ -0,0 +1,126 @@
+# coffee_maker/code_formatter/main.py
+
+import os
+import sys
+from crewai import Crew, Process
+from dotenv import load_dotenv
+from github import Github
+
+# --- 1. Load environment variables FIRST ---
+load_dotenv()
+
+# --- CORRECT LANGFUSE IMPORTS ---
+from langfuse import Langfuse
+from langfuse import observe
+from langfuse.langchain import CallbackHandler
+
+# Absolute imports from the project's source root 'coffee_maker'
+from coffee_maker.code_formatter.crewai.agents import create_code_formatter_agents, create_pr_reviewer_agent, llm
+from coffee_maker.code_formatter.crewai.tasks import create_review_task, create_refactor_task
+
+# --- 2. Initialize the global Langfuse client ---
+try:
+    langfuse_client = Langfuse(
+        secret_key=os.getenv("LANGFUSE_SECRET_KEY"),
+        public_key=os.getenv("LANGFUSE_PUBLIC_KEY"),
+        host=os.getenv("LANGFUSE_HOST"),
+    )
+except Exception as e:
+    print(f"CRITICAL ERROR: Langfuse client could not be initialized. Check environment variables.")
+    print(f"Details: {e}")
+    sys.exit(1)
+
+
+# --- HELPER FUNCTION for fetching file content from GitHub ---
+@observe
+def _get_pr_file_content(repo_full_name, pr_number, file_path):
+    try:
+        token = os.getenv("GITHUB_TOKEN")
+        if not token:
+            print("ERROR: GITHUB_TOKEN environment variable is not set.")
+            return None
+        g = Github(token)
+        repo = g.get_repo(repo_full_name)
+        pull_request = repo.get_pull(pr_number)
+        contents = repo.get_contents(file_path, ref=pull_request.head.sha)
+        return contents.decoded_content.decode("utf-8")
+    except Exception as e:
+        print(f"ERROR: Could not fetch content for '{file_path}' from GitHub. Details: {e}")
+        return None
+
+
+# --- MAIN ORCHESTRATION FUNCTION ---
+@observe
+def run_code_formatter(repo_full_name, pr_number, files_to_review, run_metadata={}):
+    """
+    Initializes and runs the crew to refactor and post suggestions for each file in a PR.
+    """
+    try:
+        handler = CallbackHandler()
+    except Exception as e:
+        print(f"ERROR: Langfuse Callback handler could not be created. Details: {e}")
```

Comment:
Removed an extra print statement and merged it with the initial error message for clarity and consistency. This makes the code cleaner and easier to understand, and it is also more efficient as it reduces the number of print statements.
```suggestion
try:
    handler = CallbackHandler()
except Exception as e:
    print(f"ERROR: Langfuse Callback handler could not be created. Details: {e}")
    return
```